### ISSUE TITLE
ZeroFloatDivision in AdaptativeMap.adapt()

### ISSUE THREAD

**Issue by kalaspa** [gplepage/vegas#14]
Hello,
First, let me thank the maintainers for this wonderful module!
I am encountering an issue when evaluating a 4-dimensional, quickly oscillating integral. Unfortunately I cannot share the source code, but here are the characteristics and the error message.

- NEVAL = 1e4
- NITNS = 10

Usually everything works perfectly, but from time to time I receive the following error
```bash
  File "_vegas.pyx", line 1622, in vegas._vegas.Integrator.__call__
  File "_vegas.pyx", line 496, in vegas._vegas.AdaptiveMap.adapt
ZeroDivisionError: float division
```
From the error message, I suspect that there is a bug in `_vegas.pyx` (or a non-documented feature) in the function `adapt()` of the adaptative map.
Here it an extract of the source code, from l. 479 to 496
```python
if self.sum_f is not None and alpha != 0:
    for i in range(old_ninc):
        avg_f[i] = self.sum_f[d, i] / self.n_f[d, i]
if alpha > 0 and old_ninc > 1:
    tmp_f[0] = (7. * avg_f[0] + avg_f[1]) / 8.
    tmp_f[-1] = (7. * avg_f[-1] + avg_f[-2]) / 8.
    sum_f = tmp_f[0] + tmp_f[-1]
    for i in range(1, old_ninc - 1):
        tmp_f[i] = (6. * avg_f[i] + avg_f[i-1] + avg_f[i+1]) / 8.
        sum_f += tmp_f[i]
    if sum_f > 0:
        for i in range(old_ninc):
            avg_f[i] = tmp_f[i] / sum_f + TINY
    else:
        for i in range(old_ninc):
            avg_f[i] = TINY
    for i in range(old_ninc):
        avg_f[i] = (-(1 - avg_f[i]) / log(avg_f[i])) ** alpha
```
It looks like `avg_f[i]` is equal to 1 for one value of `i`. I don't know how it is possible, and it is maybe a pathological behavior due to my integrand, but you might be interested in this.

Python version = 3.9.2
Vegas version = 3.5.3

**Comment by gplepage** [comment]
Thanks for posting the problem. This should never happen, no matter how pathological the integrand. So I need to fix something. I am not sure what yet, but want to check whether the number of iterations and the number of evaluations are the only parameters that you set in the integrator (ie, all other parameters have their default values). Also did this happen only once with your code is or is it easy to get it to happen again? Does it happen for larger values of neval?

Thanks again.

**Comment by kalaspa** [comment]
The only parameters I had set in the integrator at initialization were:
- A 4-dimensional map
- `mpi=False`

Then at evaluation, I had only given:
- The integrand
- Number of iterations ~ 10 for both the training and the evaluation
- Number of evaluations ~ 1e4

My use-case is about a hundred evaluations of integrals at each submission. The error occurred systematically at least twice per submission (error rate of more than 1% if you wish). Once I increased the number of evaluations, this error nearly disappeared.
